solver/src/main/java/org/chocosolver/util/objects/IdentityToDouble.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * This file is part of choco-solver, http://choco-solver.org/
+ *
+ * Copyright (c) 2025, IMT Atlantique. All rights reserved.
+ *
+ * Licensed under the BSD 4-clause license.
+ *
+ * See LICENSE file in the project root for full license information.
+ */
+package org.chocosolver.util.objects;
+
+import org.chocosolver.solver.Identity;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Define a way to store and manipulate weights associated to identifiable objects.
+ * <br/>
+ *
+ * @param <I> the type of identifiable objects
+ * @author Charles Prud'homme
+ * @implNote The values are stored in an array, indexed by the id of the identifiable object.
+ * @since 26/03/2025
+ */
+public class IdentityToDouble<I extends Identity> {
+
+    // the weights
+    private double[] values;
+
+    /**
+     * Create a structure to store weights associated with identifiable objects
+     */
+    public IdentityToDouble() {
+        this.values = new double[16];
+    }
+
+    /**
+     * Ensure that the internal array has a capacity of at least {@code capacity}
+     *
+     * @param capacity the capacity to ensure
+     */
+    private void ensureCapacity(int capacity) {
+        if (values.length < capacity) {
+            values = Arrays.copyOf(values, capacity);
+        }
+    }
+
+    /**
+     * Return the weight associated with {@code i}
+     *
+     * @param i an identifiable object
+     * @return the weight associated with {@code i}
+     */
+    public double get(I i) {
+        ensureCapacity(i.getId() + 1);
+        return values[i.getId()];
```

Comment:
It might be interesting to replace by this code (to avoid modifying/enlarging `values` just to get 0) ?

`if (values.length < i.getId() + 1) {
    return 0.;
} else {
    return values[i.getId()];
}`